scripts/test-projects/src/lib.rs

```diff
@@ -6,6 +7,50 @@ use std::{
 };
 use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
 
+const NUM_CONCURRENT: usize = 1;
+
+#[derive(Parser)]
+#[command(name = "test-projects", version, about, propagate_version = true)]
+pub struct Cli {
+    #[command(subcommand)]
+    pub command: Commands,
+
+    /// Number of concurrent projects
+    #[arg(short, long, default_value_t = NUM_CONCURRENT)]
+    #[arg(value_name = "NUM")]
+    pub num_concurrent: usize,
+
+    /// Specify where to find `forc` and `forc-fmt`
+    #[arg(long, value_name = "DIR")]
+    pub bin_path: Option<PathBuf>,
+
+    /// Specify test projects path
+    #[arg(long, value_name = "DIR")]
+    pub projects_path: Option<PathBuf>,
+}
+
+#[derive(Subcommand)]
+pub enum Commands {
+    /// Builds all test projects with `forc`
+    Build {
+        /// Cleans `forc` build output
+        #[arg(long)]
+        clean: bool,
+    },
+    /// Formats all test projects with `forc-fmt`
+    Format {
+        /// Checks format but doesn't modify files
+        #[arg(long)]
+        check: bool,
+    },
+}
+
+pub struct Command2Run {
```

Comment:
I'd call it `RunnableCommand` or `CommandInvocationData` or `PreparedCommand`. 

At the very least `CommandToRun`.